Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -378,6 +379,39 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
 
         return indexPath
       }
+
+    // FB CAPI
+    let userEmail = storedCardsEvent.values()
+      .filter(second >>> isFalse)
+      .map(first)
+      .skipNil()
+      .map { $0.me.email }
+
+    self.notifyFacebookCAPIUserEmail = userEmail.signal
+
+    _ = Signal.combineLatest(project, self.viewDidLoadProperty.signal.ignoreValues())
+      .takeWhen(didTapToAddNewCard)
```

Comment:
So I know I suggested the `goToAddCardViaStripeScreen` as the signal to use here. But rethinking this, if we follow that signal out to the view and see `goToPaymentSheet` as the signal the _eventually_ shows the Stripe payment sheet. The network request might fail here `private func goToPaymentSheet`. So use a new signal input from the `.success` case of that closure (just like `shouldCancelPaymentSheetAppearance` in the failure case). And upon receipt of the signal do this: 
```
    _ = Signal.combineLatest(project, self.viewDidLoadProperty.signal.ignoreValues())
      .takeWhen(stripePaymentSheetDidAppear)
```

That way you guarantee you send the tracking event only if the user actually sees the payment sheet. (For example, with no network, it could fail to appear)

Reply:
definitely makes sense. thanks for the helpful context here. not super familiar with the payments flow just yet